huaweicloud/services/evs/resource_huaweicloud_evs_snapshot_group.go

```diff
@@ -0,0 +1,517 @@
+			}
+			// Use JSON array format for ids parameter
+			idsJSON, err := json.Marshal(volumeIDStrs)
+			if err != nil {
+				return false, fmt.Errorf("error marshaling volume IDs: %s", err)
+			}
+			queryParams += fmt.Sprintf("&ids=%s", string(idsJSON))
+		}
+	}
+
+	queryParams += "&status=creating"
+	requestPath := client.Endpoint + "v2/{project_id}/cloudvolumes/detail"
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	if queryParams != "" {
+		requestPath += "?" + queryParams[1:]
+	}
+
+	requestOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	resp, err := client.Request("GET", requestPath, &requestOpt)
+	if err != nil {
+		return false, err
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return false, err
+	}
+
+	volumes := utils.PathSearch("volumes", respBody, []interface{}{}).([]interface{})
+
+	if len(volumes) > 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func refreshSnapshotGroupStatus(client *golangsdk.ServiceClient, snapshotGroupId string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		respBody, err := GetSnapshotGroupDetail(client, snapshotGroupId)
+		if err != nil {
+			return nil, "ERROR", err
+		}
+		status := utils.PathSearch("snapshot_group.status", respBody, "").(string)
+		if status == "" {
+			return respBody, "ERROR", errors.New("status is not found in API response")
+		}
+		return respBody, status, nil
+	}
+}
+
+func handleMultiOperationsError(err error) (bool, error) {
+	if err == nil {
+		return false, nil
+	}
+	if errCode, ok := err.(golangsdk.ErrDefault400); ok {
+		var apiError interface{}
+		if jsonErr := json.Unmarshal(errCode.Body, &apiError); jsonErr != nil {
+			return false, fmt.Errorf("unmarshal the response body failed: %s", jsonErr)
+		}
+		errorCode, errorCodeErr := jmespath.Search("error_code||errCode", apiError)
+		if errorCodeErr != nil {
+			return false, fmt.Errorf("error parse errorCode from response body: %s", errorCodeErr)
+		}
+		if errorCode == "EVS.2409" {
+			return true, err
+		}
+	}
+	if errCode, ok := err.(golangsdk.ErrUnexpectedResponseCode); ok && errCode.Actual == 409 {
+		return true, err
+	}
+	return false, err
+}
+
+func resourceEvsSnapshotGroupRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v5/{project_id}/snapshot-groups/{snapshot_group_id}"
+		product = "evs"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating EVS client: %s", err)
+	}
+	requestPath := client.Endpoint + httpUrl
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	requestPath = strings.ReplaceAll(requestPath, "{snapshot_group_id}", d.Id())
+	requestOpt := golangsdk.RequestOpts{KeepResponseBody: true}
+	resp, err := client.Request("GET", requestPath, &requestOpt)
```

Comment:
You have already define function GetSnapshotGroupDetail(), why not just use it?
```suggestion
	respBody, err := GetSnapshotGroupDetail(client, d.Id())
```